spylon/spark/launcher.py

```diff
@@ -395,8 +399,13 @@ def __getattr__(self, key):
         if key.startswith("_"):
             return super(SparkConfiguration, self).__getattribute__(key)
         spark_arg = key.replace('_', '-')
-        if spark_arg in self._spark_launcher_arg_names:
-            return self._spark_launcher_args.get(spark_arg)
+        attr_err = '%s object has no attribute %s' % (self.__class__.__name__, key)
+        if spark_arg not in self._spark_launcher_arg_names:
+            raise AttributeError(attr_err)
+        try:
+            return self._spark_launcher_args[spark_arg]
```

Comment:
Should this still use `.get()`? I can see that if the argument name is not known, then AttributeError is the correct thing to raise. However, if the attribute name is known (i.e., it's in _spark_launcher_arg_names, but the attribute is simply not set yet, then None is a better sentinel in my mind. Otherwise, it's impossible to distinguish known-attributes that are not yet set from unknown-attributes that can never be set.